native/cocos/renderer/pipeline/custom/FrameGraphDispatcher.cpp

```diff
@@ -896,27 +841,28 @@ struct PassVisitor : boost::dfs_visitor<> {
     CloseCircuits &_circuits;
 };
 
+// forward (vertex ascending):
+// -- true: how much resource this pass writes to, which has an effect of later passes;
+// -- false: how much resource this pass reads from, which is dependent from former passes.
 auto evaluateHeaviness(const RAG &rag, const ResourceGraph &rescGraph, EmptyVert vert, bool forward) {
-    // forward:
-    // -- true: how much resource this pass writes to, which has an effect of later passes;
-    // -- false: how much resource this pass reads from, which is dependent from former passes.
     const ResourceAccessNode &accessNode = get(RAG::AccessNode, rag, static_cast<RAG::vertex_descriptor>(vert));
     int64_t score = 0;
     bool forceAdjacent = false;
-    for (const auto &resc : accessNode.attachemntStatus) {
+    for (const auto &resc : accessNode.attachmentStatus) {
         int64_t eval = 0;
         auto rescID = resc.vertID;
         const ResourceDesc &desc = get(ResourceGraph::Desc, rescGraph, rescID);
         const ResourceTraits &traits = get(ResourceGraph::Traits, rescGraph, rescID);
 
-        gfx::MemoryAccessBit filter = forward ? gfx::MemoryAccessBit::WRITE_ONLY : gfx::MemoryAccessBit::READ_ONLY;
-        if (resc.access == filter) {
+        gfx::MemoryAccessBit substractFilter = forward ? gfx::MemoryAccessBit::READ_ONLY : gfx::MemoryAccessBit::WRITE_ONLY;
+        if (resc.access == substractFilter) {
+            // forward calculate write(s), backward calculate read(s).
             continue;
         }
 
         switch (desc.dimension) {
             case ResourceDimension::BUFFER:
-                eval = desc.width;
```

Comment:
Is this change right?

Reply:
ah.. no, if i understand correctly `desc.width` comes to buffer size, so this should still be `desc.width`?

Reply:
Yes, width is buffer size.